client/src/main/java/io/split/client/SplitFactoryImpl.java

```diff
@@ -634,9 +634,11 @@ private ImpressionsManagerImpl buildImpressionsManager(SplitClientConfig config,
         ImpressionListener listener = !impressionListeners.isEmpty()
                 ? new ImpressionListener.FederatedImpressionListener(impressionListeners)
                 : null;
+        counter = new ImpressionCounter();
```

Comment:
Can we move this when it is NONE case?

Reply:
We have to create the None strategy by default now, which is why we have the counter object as well.

Reply:
Good, I got it. Thanks Bilal.